server/commands/write_authzmodel.go

```diff
@@ -163,3 +185,83 @@ func isUsersetRewriteValid(allRelations map[string]struct{}, relationsOnType map
 
 	return nil
 }
+
+func areRelationalTypesValid(tds []*openfgapb.TypeDefinition) error {
+	typeToRelations := map[string]map[string]struct{}{}
+	for _, td := range tds {
+		typeName := td.GetType()
+		typeToRelations[typeName] = map[string]struct{}{}
+		for relationName := range td.GetRelations() {
+			typeToRelations[typeName][relationName] = struct{}{}
+		}
+	}
+
+	// Here we are checking that every type on a relation is valid. This means:
+	// 1. If it is a type (e.g. user) then this type is a type in the model
+	// 2. If it is a type#relation then this type with this relation is in the model
+	for _, td := range tds {
+		for _relation, _ := range td.GetRelations() {
+			metadata, ok := td.GetMetadata().GetRelations()[_relation]
+			if ok {
+				for _, userType := range metadata.GetDirectlyRelatedUserTypes() {
+					objectType := userType.GetType()
+					relation := userType.GetRelation()
+
+					if _, ok := typeToRelations[objectType]; !ok {
+						return serverErrors.InvalidRelationType(td.GetType(), _relation, objectType, relation)
```

Comment:
i think these two errors should be different, because at this point, `relation` can be empty

Reply:
It is the type of the relation that is invalid so I think this makes sense. Suppose the type is `user` and then:

1. Without relation the error is "the type "user" is invalid on ..."
2. With a relation, say `member`, the error is "the type "user#member` is invalid on ..."

WDYT?

Reply:
You mean that you would use `InvalidRelationType` in both but format the error nicely? If so, 👍